src/mon/OSDMonitor.cc

```diff
@@ -3097,7 +3098,7 @@ void OSDMonitor::check_osdmap_sub(Subscription *sub)
 
 void OSDMonitor::check_pg_creates_subs()
 {
-  if (!osdmap.get_num_up_osds()) {
+  if (osdmap.get_num_up_osds() == 0) {
```

Comment:
Will it make any difference?
changing ! to ==

Reply:
Should not make not any difference on its behavior. However we should use same expression as much as possible :( 

https://github.com/ceph/ceph/blob/b4e4ad3b855f86d2d6f7d07c670c79e3dab59655/src/mon/OSDMonitor.cc#L1369